node-website-feed/README.md

```diff
@@ -11,3 +13,147 @@ Bare bones off the shelf website feed for NodeJS, including Client Credentials a
 5. Re-name the `src/.env.example` file to `.env` and add your `clientId` and `clientSecret`
 6. `yarn dev` or `npm run dev` to run a development server (`yarn start` or `npm run start` in prod)
 7. Load your browser at `http://localhost:3000`
+
+### Suggested Architecture
+
+In the code example, we fetch a list of properties on the fly. This is fine if you need the data to be the latest in our DB at the exact moment of execution however, practically for serving a website, it is expensive in terms of processing on the server, http round trips and financial cost by API calls.
+
+So how do I build on the website feed example?
+
+The simplest reasonable design pattern is as follows;
+
+#### 1. **Seed your application**
+
+Fetch all properties and embedded property images in a paged response, and commit to a database of your choosing. The query will be paginated at a maximum of 100 so you will need to do this step in a few requests, depending on your client.
+
+The following query will fetch all properties where;
+
+- The selling status is `forSale` - you can specify other statuses as you wish eg `underOffer`
+- The letting status is `toLet` and `currentlyAvailable` - as above, you can specify additional statuses
+- The customer has specified `internetAdvertising` is true
+- The page size is 100, and the page number is 1 - you will need to increment the page number until no more properties match your query
+- Embed property images. This will return all property images associated to each property. You should store these in a separate location in your database.
```

Comment:
We probably need to discuss if it's appropriate to link directly to assets on our network. The images API returns a public URL. If a developer links directly to our assets, if our systems go down so does the website. There's probably arguments for/against doing this to be honest though. We've had issues in the past with people linking directly to our stuff but in my view we should allow it - others might disagree though as we can't really protect ourselves against search engines visiting customer websites and hammering our servers

Reply:
I see what you are saying but probably out of scope here unless we have a better suggestion now?

Reply:
Fine to leave it, might be something we have to revisit at some point but let's see